### ISSUE TITLE
TCH001 false positive in Ruff 0.0.281

### ISSUE THREAD

**Issue by flying-sheep** [astral-sh/ruff#6233]
When updating to Ruff 0.0.281, the following started falsely throwing TCH001:

[playground 1](https://play.ruff.rs/#N4KABGBEDOCmA2sDGAXSAuMBtSBBAMvpALoA04UKAhgE4DmsKAtAG6w3QCWA9gHYZQADgE8AzAEZxkEAF8AJAshKUsaCgB0S6SABmNbgFswAfWM6ArinM1YpsJwODuNFGCq9e3aih69oIXX0jFGFBTl46e0dnVwAVAE0ABQBRYwBhAAlktIBpAEkAOQBxAL1DMHVjaHNBJxcoutdq2pjYABNjGmMDKhQaTgAPYyR4Kmg4fxBOHTAElPSs3MKi9AoIMqNoJE4RBpiwaEFaOACQUyp4eDsAXmw1ym4kaBpIcggIGBrG9s7u3v6hiMxhNXiBiKc2rAZihHs8ABTcABGACtMIdjrB1Icen1BgBKMBMAB8ByOHExTy6OIBq3eUGUqg0WnuNisNF4YCRyPUMOG8LxICAA) | [playground 2](https://play.ruff.rs/#N4KABGBEDOCmA2sDGAXSAuMBtSBBAMvpALoA04UKAhgE4DmsKAtAG6w3QCWA9gHYZQADgE8AzAEZxkEAF8AJAshKUsaCgB0S6SABmNbgFswAfWM6ArinM1YpsJwODuNFGCq9e3aih69oIXX0jQSoUAAt4TgAje0dnVwAFULDAwzAUYUFOXjpYpxcwABUATQSAUWMAYQAJMsqAaQBJADkAcQDOHSLSipq6prb0Cgg9NPdeABNQqjz4sFwPABFp4bBR4Pcp6FmC5eoAMRoqA1gA0yp4eDsAXmxIUzYOX1NIUigADxIA2oAlMrBbklwgAKUw6TiIUwASnUIRsvBQARQNGEQwgIyCYDgVkEKG43Hg0GM0CQRgc+VcDBQxkeXD4AXRJhp7DpvBuYCpzKefGBahowN+ZVhtFgCOF8JQUKhIFg7yQsFxYGBjTiLjKNH0NDe+HxAGtzIJ1ZqoWj0cjUatGeswOouaydq4HiznsZVrL5YqVRSjc43NtYKbGRADNBcrdIJVxl57LwdOwwLTfOpNJb0UdOHAwF74j7+SG6FC1pjTgEJrAuu9gVQJpgFrw9lRC0wAHxgBuHY4B1ZaFRqTRKVY2Kw0XhuCbqPFmI4nYHSoA)

```py
"""test."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ._support import supported_r_matrix_classes

if TYPE_CHECKING:
    from scipy import sparse


__all__ = [
    "tocsr",
    "supported_r_matrix_classes",
]


def tocsr(obj: sparse.spmatrix) -> sparse.csr_matrix:
    """test."""
    return obj.to_csr()
```

```
    from ._support import supported_r_matrix_classes
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
TCH001: Move application import `._support.supported_r_matrix_classes` into a type-checking block
```

**Comment by mkniewallner** [comment]
Probably the same underlying issue as https://github.com/astral-sh/ruff/issues/6207, which should be fixed in https://github.com/astral-sh/ruff/pull/6208.

**Comment by flying-sheep** [comment]
indeed, duplicate of #6207

**Comment by charliermarsh** [comment]
This should be fixed in v0.0.282, which is out now.

**Comment by charliermarsh** [comment]
I'm also going to add Poetry to our ecosystem CI checks, to prevent future regressions.
